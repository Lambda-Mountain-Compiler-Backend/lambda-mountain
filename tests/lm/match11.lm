
f := Î»p . (
   (match p (
      ()
      ( () p )
      ( Nil p )
      ( (Variable _) p )
      ( (Literal _) p )
      ( (App( g x )) (
         (App( (f g) (f x) ))
      ))
      ( (Lambda( lhs rhs )) (
         (App( (f lhs) (f rhs) ))
      ))
      ( (prog (GExpr( g ))) (
         ( (f prog) (GExpr (f g)) )
      ))
      ( (prog (Global( n v ))) (
         ( (f prog) (Global( (f n) (f v) )) )
      ))
      ( (prog (Type( tn td ))) (
         ( (f prog) (Type( (f tn) (f td) )) )
      ))
      ( unknown (
         (UnknownTerm unknown)
      ))
   ))
);

main := print-s ( f (() (Global (S V))) );
