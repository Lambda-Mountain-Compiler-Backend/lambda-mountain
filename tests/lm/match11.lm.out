(() (Global ((UnknownTerm S) (UnknownTerm V))))
